android/src/main/java/com/imagepicker/ImagePickerModule.java

```diff
@@ -220,20 +221,17 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
     : data.getData();
 
     String realPath = getRealPathFromURI(uri);
-
-    boolean isUrl = true;
-    try {
-        URL url = new URL(realPath);
-    }
-    catch (MalformedURLException e) {
-        isUrl = false;
-    }
-    if (isUrl) {
-        // @todo handle url as well (Facebook image, etc..)
-        response.putString("uri", uri.toString());
-        response.putString("urlPath", realPath);
```

Comment:
Why remove the fix for **< 4.4** ?
You can remove it but partially and use your fix for **< 4.4** aswell
```java
String realPath = getRealPathFromURI(uri);
boolean isUrl = true;

if (realPath !=  null) {
  // test for < 4.4
  try {
    URL url = new URL(realPath);
  } catch (MalformedURLException e) {
   isUrl = false;
  }
} else {
  isUrl = false;
}

if (realPath ==  null || isUrl) {
 try {
    File file = createFileFromURI(uri);
    realPath = file.getAbsolutePath();
    uri = Uri.fromFile(file);
  }
  catch(Exception e) {
     response.putString("error", "Could not read photo");
     mCallback.invoke(response);
     return;
   }
}
```